packages/core/core/src/types.ts

```diff
@@ -718,3 +741,29 @@ export type ApiClientMethods<T> = {
     (...args: [...Parameters<T[K]>, CustomQuery?]) => ReturnType<T[K]> :
     T[K]
 }
+
+export interface UseStoreErrors {
+  load: Error | null;
+  change: Error | null;
+}
+
+export interface UseStoreFactoryParams<STORES, CHANGE_PARAMS> extends FactoryParams {
+  load(context: Context, params: { customQuery: CustomQuery; }): Promise<STORES>
+  change(context: Context, parmas: { current: STORES; next: CHANGE_PARAMS; customQuery?: CustomQuery; }): Promise<STORES>
+}
+export interface UseStoreInterface<STORES, CHANGE_PARAMS> {
+  change(params: CHANGE_PARAMS & { customQuery?: CustomQuery; }): Promise<void>;
+  load(params?: { customQuery: CustomQuery; }): Promise<void>;
+  loading: ComputedProperty<boolean>;
+  response: ComputedProperty<STORES>;
+  error: ComputedProperty<UseStoreErrors>;
+}
+
+export interface UseStore<STORES, CHANGE_PARAMS> {
+  (): UseStoreInterface<STORES, CHANGE_PARAMS>;
+}
+
+export interface UseStoreGetters<STORES, STORE_ITEM> {
+  getItems(stores: STORES, ...xs: any[]): STORE_ITEM[];
+  getSelected(stores: STORES): STORE_ITEM
```

Comment:
I would suggest using AgnosticStore[] and AgnosticStore here, without generics - I know you typed also private fields (prefixed by `_`) but... they don't have to be typed, it's something for our internal usage and end developer doesn't have to be aware of it